src/glossary.md

```diff
@@ -125,17 +140,32 @@ Strings slices are always valid UTF-8.
 
 ### Trait
 
-A trait is a language item that is used for describing the functionalities a type must provide.
-It allows a type to make certain promises about its behavior.
+A trait is a language item that is used for describing the functionalities a
+type must provide. It allows a type to make certain promises about its behavior.
 
-Generic functions and generic structs can use traits to constrain, or bound, the types they accept.
+Generic functions and generic structs can use traits to constrain, or bound,
+the types they accept.
+
+[More][traits].
+
+### Trait Implementation
```

Comment:
The wording here can be challenging for newcomers.  The confusion comes primarily(in my case), from the following two places: "associates items through a trait" and "the trait bound of the trait".  With multiple reads, I do understand what was intended.  

In the trait definition "promises" and "behaviour" are referenced, so something along the lines of the following may be easier to understand, with minimal information loss:

An implementation that associates items that will fulfill the promises a given trait makes.

An implementation that associates the items that fulfill the the behaviour requirements of a given trait.